internal/repository/file.go

```diff
@@ -0,0 +1,307 @@
+package repository
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/storage"
+	"fyne.io/fyne/storage/repository"
+)
+
+// declare conformance with repository types
+var _ repository.Repository = (*FileRepository)(nil)
+var _ repository.WriteableRepository = (*FileRepository)(nil)
+var _ repository.HierarchicalRepository = (*FileRepository)(nil)
+var _ repository.ListableRepository = (*FileRepository)(nil)
+var _ repository.MovableRepository = (*FileRepository)(nil)
+var _ repository.CopyableRepository = (*FileRepository)(nil)
+
+var _ fyne.URIReadCloser = (*file)(nil)
+var _ fyne.URIWriteCloser = (*file)(nil)
+
+type file struct {
+	*os.File
+	uri fyne.URI
+}
+
+func (f *file) URI() fyne.URI {
+	return f.uri
+}
+
+// FileRepository implements a simple wrapper around golang's filesystem
+// interface libraries. It should be registered by the driver on platforms
+// where it is appropriate to do so.
+//
+// This repository is suitable to handle file:// schemes.
+//
+// Since: 2.0.0
+type FileRepository struct {
+}
+
+// NewFileRepository creates a new FileRepository instance. It must be
+// given the scheme it is registered for. The caller needs to call
+// repository.Register() on the result of this function.
+//
+// Since: 2.0.0
+func NewFileRepository() *FileRepository {
+	return &FileRepository{}
+}
+
+// Exists implements repository.Repository.Exists
+//
+// Since: 2.0.0
+func (r *FileRepository) Exists(u fyne.URI) (bool, error) {
+	p := u.Path()
+
+	_, err := os.Stat(p)
+	ok := false
+
+	if err == nil {
+		ok = true
+	} else if os.IsNotExist(err) {
+		err = nil
+	}
+
+	return ok, err
+}
+
+func openFile(uri fyne.URI, create bool) (*file, error) {
+	if uri.Scheme() != "file" {
+		return nil, fmt.Errorf("invalid URI for file: %s", uri)
+	}
+
+	path := uri.String()[7:]
+	var f *os.File
+	var err error
+	if create {
+		f, err = os.Create(path) // If it exists this will truncate which is what we wanted
+	} else {
+		f, err = os.Open(path)
+	}
+	return &file{File: f, uri: uri}, err
+}
+
+// Reader implements repository.Repository.Reader
+//
+// Since: 2.0.0
+func (r *FileRepository) Reader(u fyne.URI) (fyne.URIReadCloser, error) {
+	return openFile(u, false)
+}
+
+// CanRead implements repository.Repository.CanRead
+//
+// Since: 2.0.0
+func (r *FileRepository) CanRead(u fyne.URI) (bool, error) {
+	f, err := os.OpenFile(u.Path(), os.O_RDONLY, 0666)
+	if err == nil {
+		defer f.Close()
+	} else {
+
+		if os.IsPermission(err) {
+			return false, nil
+		}
+
+		if os.IsNotExist(err) {
+			return false, nil
+		}
```

Comment:
If InMemoryRepository returns an error for non-existent resources, then so should FileRepository. Although I think it is best that neither return an error, and simply `return false, nil` as this appropriately answers the question `CanRead`